### ISSUE TITLE
debian packaging

### ISSUE THREAD

**Issue by anarcat** [la5nta/pat#95]
hi again :)

for your information, i started looking at packaging pat for debian. the golang packaging team has shown interest in helping, if not in directly packaging it, at least acting as an umbrella team for the packaging. i have also notified the debian-hams team to raise interest on the project.

you can follow the progress of the package in [#877030](https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=877030) on the Debian BTS. if you are interested in maintaining the package yourself upstream, that's also a possibility, especially if you are a debian maintainer or developer. if not, I can also act as a sponsor when a new release is published.

this is just a notification, but can also act as a coordination point for upstream packaging if you wish to followup on this. otherwise, feel free to close this issue whenever you like.

thank you for your attention.

a.

**Comment by martinhpedersen** [comment]
This is great - I would love to see Pat in the Debian repos! Thank you for taking initiative.

I've considered this previously but concluded that I should wait until at least the v1 release, mainly due to the limited active developers we have.

I have subscribed to the debian bug and will monitor closely 👍

**Comment by martinhpedersen** [comment]
If you can act as a sponsor, that would be greatly appreciated. What do you need me to do?

**Comment by anarcat** [comment]
> I've considered this previously but concluded that I should wait until at least the v1 release, mainly due to the limited active developers we have.

understandable. keep in mind that, if you do not want to see a development release of yours shipped in a debian stable release that will be supported for years, there are ways to do that: either upload only to experimental, or file an RC bug against the debian package to keep it from migrating.

> I have subscribed to the debian bug and will monitor closely 👍

woot. :)

> If you can act as a sponsor, that would be greatly appreciated. What do you need me to do?

well, the basics is that we need to have a working "debian package" here that conforms to policy. make sure the package builds properly and passes all lintian tests. we also need to package dependencies - as you've seen in the Debian BTS, there's one dependency that needs to be uploaded before pat makes it into debian (which is really good - often we have tens of dependencies to fix to get certain packages in). you probably will want a package for the w2lk library as well.

once that's all done, upload the results to https://mentors.debian.net and ping me or file a "RFS" (Request For Sponsorship) on the Debian BTS, if I disappeared. you can also contact the debian-golang or debian-hams teams for help, i'm sure they'd be happy to sponsor a well-written package.

**Comment by martinhpedersen** [comment]
I'm sorry anarcat, I may have misunderstood you earlier. If you are offering to coordinate with the golang packaging team and package Pat for debian, it would highly appreciated.

Was your initial comment an offer to do so? I am sorry for the confusion.

(I will of course assist in any way and accept any PR to make the task easier.)

**Comment by anarcat** [comment]
On 2017-09-29 14:07:39, Martin Hebnes Pedersen wrote:
> I'm sorry anarcat, I may have misunderstood you earlier. If you are offering to coordinate with the golang packaging team and package Pat for debian, it would highly appreciated.
>
> Was your initial comment an offer to do so? I am sorry for the confusion.
>
> (I will of course assist in any way and accept any PR to make the task easier.)

i can do whatever, of course. :) the trick is that you have a debian/
directory already in there, which usually means upstream is working on
the debian package somehow.

the more common practice when upstream *isn't* the maintainer is to
*not* have a debian/ directory upstream, and keep that managed on
Debian's side. in effect, it means that the debian package is different
branch that only has an extra debian/ directory in it (although some
people make patches to the upstream code directly in git as well).

in other words:  either we maintain the package here on github or we
maintain it on debian's infrastructure. :)

i'm easy with anything, but i will need push access to the debian/ tree
for this to make sense if i'm the maintainer.

thanks!

-- 
The destiny of Earthseed is to take root among the stars.
                        - Octavia Butler

**Comment by anarcat** [comment]
oh and for what it's worth, i'll probably be using this magic tool to package the dependencies: https://people.debian.org/%7Estapelberg/2015/07/27/dh-make-golang.html

**Comment by martinhpedersen** [comment]
>i can do whatever, of course. :)

That sounds fantastic! I'm sure it will be appreciated by many users, thank you.

>the trick is that you have a debian/ directory already in there, which usually means upstream is working on the debian package somehow.

I haven't got any strong feelings about keeping the debian/ directory upstream or not.

However I am a little concerned about the availability and portability of the packages if we merge the current debian/ directory with the changes needed to get it into Debian.

In particular;
* The current .deb packages is statically linked against the lastest upstream libax25. See issue #59.
* It's very easy for users to simply download the .deb files and keep a local copy for field deployment.
* We need to push releases far more often than the debian release cycle supports.
  The WDT (Winlink Development Team) expects client software to implement support for potentially breaking API changes very rapidly.

For these reasons, I believe we should continue to provide the current .deb packages in addition to pushing releases to the debian repos.

What is your thoughts on this?

>i'm easy with anything, but i will need push access to the debian/ tree
>for this to make sense if i'm the maintainer.

I will be happy to grant you push access to the repo if we decide it's the best way forward :)

**Comment by martinhpedersen** [comment]
> oh and for what it's worth, i'll probably be using this magic tool to package the dependencies: https://people.debian.org/%7Estapelberg/2015/07/27/dh-make-golang.html

Yes, I've seen that one before. It would be very interesting if you could share your experience regarding this when you get around to it.

PS: github.com/bndr/gotabulate is not a very important dependency, and we can probably get by without if that's what it takes.

**Comment by DC7IA** [comment]
I've done some research and found this.

https://pmateusz.github.io/linux/2017/06/30/linux-secure-apt-repository.html

You can use GitHub pages to host a Debian repo.

https://assafmo.github.io/2019/05/02/ppa-repo-hosted-on-github.html

You can use a GitHub repo to host a Debian repo.


Maybe one of these options?

Then the repo key just needs to be places on getpat.io, since the source of a package can't be the source for verification.

**Comment by DC7IA** [comment]
Debian Testing now includes pat:

https://packages.debian.org/bookworm/pat

It seems Debian packaging is already done, but we have to wait for the next release.

Should we close this and create an issue for Ubuntu or instead leave this open..?

**Comment by DC7IA** [comment]
Of course we still need a Debian repo for Ubuntu and Linux Mint.

I don't know whether they will also include pat. :thinking: 

Does Ubuntu import all Debian packages..?
